### PR TITLE
[ascend] fix prefix caching

### PR DESCRIPTION
**Pull request by yao-fengchen**

### CODE REVIEW

File: lmdeploy/pytorch/backends/dlinfer/ascend/op_backend.py [hunk 1]
```diff
@@ -389,11 +385,11 @@ def get_moe_group_name(group):
             q_seqlens=q_seqlens_cpu,
             # kv_seqlens_expanded is only expanded in paged prefill,
             # otherwise it equals kv_seqlens_cpu
```
Comment (jinminxi104) [inline lmdeploy/pytorch/backends/dlinfer/ascend/op_backend.py]: remove these unused comments

File: lmdeploy/pytorch/backends/dlinfer/ascend/op_backend.py [hunk 2]
```diff
@@ -187,37 +185,43 @@ def get_cpu_seqlens(is_decoding, is_unpaged_prefill):
...
-        def get_kv_start_indices_and_attention_mask(is_decoding, is_unpaged_prefill, q_seqlens_list, kv_seqlens_list,
+        def update_q_seqlens(is_decoding, is_prefill_no_cache, q_seqlens_cpu=None):
+            if is_decoding:
+                batch_size = step_context.q_seqlens.size(0)
+                return torch.arange(1, batch_size + 1, dtype=torch.int32)
+            elif is_prefill_no_cache:
+                return q_seqlens_cpu
+            return q_seqlens_cpu.cumsum(dim=0)
+
+        def get_kv_start_indices_and_attention_mask(is_decoding, is_prefill_no_cache, q_seqlens_list, kv_seqlens_list,
```
Comment (jinminxi104) [inline lmdeploy/pytorch/backends/dlinfer/ascend/op_backend.py:224]: q_seqlens_list, kv_seqlens_list are not used

Comment (yao-fengchen) [reply lmdeploy/pytorch/backends/dlinfer/ascend/op_backend.py:224]: > q_seqlens_list, kv_seqlens_list are not used

q_seqlens_list and kv_seqlens_list are used when calculating slots

File: lmdeploy/pytorch/backends/dlinfer/ascend/op_backend.py [hunk 3]
```diff
@@ -236,25 +240,17 @@ def get_kv_start_indices_and_attention_mask(is_decoding, is_unpaged_prefill, q_s
...
                         torch.triu(torch.ones(max_q_seq_len,
                                               max_kv_seq_len,
                                               dtype=step_context.kv_caches[0][0].dtype,
                                               device=step_context.block_offsets.device),
                                    diagonal=max_kv_seq_len - max_q_seq_len + 1))
                 else:
-                    attention_mask = [torch.cat(attention_mask)]
+                    attention_mask.append(
+                        torch.triu(torch.ones(2048, 2048, dtype=torch.bool, device=step_context.block_offsets.device),
+                                   diagonal=1))
```
[the remark this answers is not in the document]
Comment (yao-fengchen) [reply lmdeploy/pytorch/backends/dlinfer/ascend/op_backend.py:253]: > The attention mask is now hard-coded to a 2048x2048 tensor, which can produce incorrect masking (or shape mismatches) whenever `max_q_seq_len` / `max_kv_seq_len` differ from 2048. This should be derived from the actual computed lengths (e.g., `max_q_seq_len`, `max_kv_seq_len`, or `step_context.block_offsets.shape[1] * block_size`) so the mask matches the kernel’s expected dimensions.

NPU operator requires the mask to be a fixed shape (2048, 2048)

File: lmdeploy/pytorch/backends/dlinfer/ascend/op_backend.py (hunk 1 above)
Comment (yao-fengchen) [reply lmdeploy/pytorch/backends/dlinfer/ascend/op_backend.py]: done

[the remark this answers is not in the document]
Comment (yao-fengchen) [reply lmdeploy/pytorch/backends/dlinfer/ascend/op_backend.py]: done
